app/models/machine.rb

```diff
@@ -1,5 +1,13 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+
   belongs_to :owner
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+  def average
+    average = self.snacks.average(:price)
+    average.round(2)
```

Comment:
take a look at an action view helper called `number_to_currency` , that way you can take the rounding logic out of here and format it in the view.